csharp/src/Ice/SslTransceiver.cs

```diff
@@ -209,16 +209,16 @@ public async ValueTask<int> SendAsync(IList<ArraySegment<byte>> buffer, Cancella
             }
         }
 
-        private X509Certificate? CertificateSelectionCallback(
+        private X509Certificate CertificateSelectionCallback(
             object sender,
             string targetHost,
             X509CertificateCollection? certs,
-            X509Certificate remoteCertificate,
+            X509Certificate? remoteCertificate,
             string[]? acceptableIssuers)
         {
             if (certs == null || certs.Count == 0)
             {
-                return null;
+                throw new ArgumentException("X509Certificate not available", nameof(certs));
```

Comment:
It's unclear how this exception from the certificate selection callback will be handled by the SSL engine and I wonder if it's really correct. What happens if the client configures no certificate? It looks like we always set the local certificate selection callback even if the client configures no certs so this would throw in this case (aborting the SSL handshake?).

Reply:
Maybe is better to not set the selection callback in this case, and assert that the certificates collections is not null or empty

Reply:
Right, if the user doesn't explicitly set this callback and doesn't set any client certificates, there's no point in setting the callback.